server/routes/wishCards.js

```diff
@@ -554,85 +554,96 @@ router.get('/status/:id', async (req, res) => {
+  let isDonated = true;
+  const scrapeResponse = await scrapeList(`https://www.amazon.com/hz/wishlist/ls/${wishListId}`);
 
-        if (Object.keys(scrapeResponse).length > 0) {
-          isDonated = !JSON.stringify(scrapeResponse).includes(itemId);
-        }
+  if (!scrapeResponse) isDonated = false;
 
-        if (isDonated) {
-          clearTimeout(blockedWishcardsTimer[wishCardId]);
+  if (Object.keys(scrapeResponse).length > 0) {
+    isDonated = !JSON.stringify(scrapeResponse).includes(itemId);
+  }
 
-          const wishCard = await WishCardRepository.getWishCardByObjectId(wishCardId);
+  if (isDonated) {
+    clearTimeout(blockedWishcardsTimer[wishCardId]);
 
-          wishCard.status = 'donated';
-          wishCard.isLockedUntil = null;
-          wishCard.isLockedBy = null;
-          wishCard.save();
+    const wishCard = await WishCardRepository.getWishCardByObjectId(wishCardId);
 
-          await DonationsRepository.createNewDonation({
-            donationTo: wishCardId,
-            donationFrom: userId,
-            donationPrice: price,
-            donationConfirmed: true,
-          });
+    wishCard.status = 'donated';
+    wishCard.isLockedUntil = null;
+    wishCard.isLockedBy = null;
+    wishCard.save();
 
-          log.info('Wishcard donated', { type: 'wishcard_donated', wishCardId, userId });
-          io.emit('donated', { id: wishCardId, donatedBy: userId });
-          const user = await UserRepository.getUserByObjectId(userId);
-          sendDonationNotificationToSlack(user, wishCard)
+    await DonationsRepository.createNewDonation({
+      donationTo: wishCardId,
+      donationFrom: userId,
+      donationPrice: price,
+      donationConfirmed: true,
+    });
 
-          done(true);
-          return true;
-        }
+    log.info('Wishcard donated', { type: 'wishcard_donated', wishCardId, userId });
+    io.emit('donated', { id: wishCardId, donatedBy: userId });
+    const user = await UserRepository.getUserByObjectId(userId);
+    sendDonationNotificationToSlack(user, wishCard);
+
+    callback(true);
+    return true;
+  }
 
-        log.info('Wishcard not donated', { type: 'wishcard_not_donated', wishCardId, userId });
-        io.emit('not_donated', { id: wishCardId, userId });
+  log.info('Wishcard not donated', { type: 'wishcard_not_donated', wishCardId, userId });
+  io.emit('not_donated', { id: wishCardId, userId });
+
+  callback(false);
+  return true;
+}
+
+queue.process(async (job, done) => {
+  const { wishCardId, userId, url, price } = job.data;
 
-        done(false);
-        return true;
+  try {
+    const wishListArray = /registryID\.1=(.*?)&.*?registryItemID.1=(.*?)&/gm.exec(url);
+    if (wishListArray) {
+      const wishListId = wishListArray[1];
+      const itemId = wishListArray[2];
+      const wishcardInfo = { wishCardId, userId, price, itemId, wishListId };
+      if (wishListId) {
+        if (process.env.LOCAL_DEVELOPMENT) {
```

Comment:
this should be checked for the testing environment as well
@gnorbsl please correct me if I'm wrong

Reply:
Local dev is for both test and dev, I believe :thinking: